File: www/cgi-bin/hello.php
<?php
$time = date("H:i:s");
$date = date("d/m/Y");
?>
<!DOCTYPE html>
<html>
<head>
    <title>🐘 PHP Simple</title>
    <style>
        body {
            font-family: 'Arial', sans-serif;
            background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
            color: white;
            text-align: center;
            padding: 50px 20px;
            margin: 0;
        }
        .container {
            background: rgba(255,255,255,0.1);
            border-radius: 15px;
            padding: 40px;
            backdrop-filter: blur(10px);
            display: inline-block;
            min-width: 300px;
        }
        .elephant {
            font-size: 4em;
            margin-bottom: 20px;
            animation: sway 3s ease-in-out infinite;
        }
        @keyframes sway {
            0%, 100% { transform: rotate(-5deg); }
            50% { transform: rotate(5deg); }
        }
        h1 {
            margin: 0 0 20px 0;
            color: #ffd700;
        }
        .info {
            background: rgba(255,255,255,0.2);
            border-radius: 10px;
            padding: 15px;
            margin: 15px 0;
        }
        a {
            color: #ffd700;
            text-decoration: none;
            font-weight: bold;
        }
    </style>
</head>
<body>
    <div class="container">
        <div class="elephant">🐘</div>
        <h1>¡PHP funcionando!</h1>
        
        <div class="info">
            📅 Fecha: <?php echo $date; ?>
        </div>
        
        <div class="info">
            🕐 Hora: <?php echo $time; ?>
        </div>
        
        <div class="info">
            🌐 Método: <?php echo $_SERVER['REQUEST_METHOD'] ?? 'GET'; ?>
        </div>
        
        <p><a href="/cgi.html">🔙 Volver</a></p>
    </div>
</body>
</html>

File: www/cgi-bin/hello.py
#!/usr/bin/python3

import os
import datetime

print("Content-Type: text/html")
print("")

print("""<!DOCTYPE html>
<html>
<head>
    <title>🐍 Hola desde Python!</title>
    <meta charset="UTF-8">
    <style>
        body {
            font-family: 'Arial', sans-serif;
            background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
            margin: 0;
            padding: 20px;
            color: white;
        }
        .container {
            max-width: 600px;
            margin: 0 auto;
            background: rgba(255,255,255,0.1);
            border-radius: 15px;
            padding: 30px;
            backdrop-filter: blur(10px);
        }
        h1 {
            text-align: center;
            font-size: 2.5em;
            margin-bottom: 30px;
            text-shadow: 2px 2px 4px rgba(0,0,0,0.3);
        }
        .info-box {
            background: rgba(255,255,255,0.2);
            border-radius: 10px;
            padding: 20px;
            margin: 20px 0;
        }
        .emoji {
            font-size: 2em;
            margin-right: 10px;
        }
        a {
            color: #ffd700;
            text-decoration: none;
            font-weight: bold;
        }
        a:hover {
            text-decoration: underline;
        }
    </style>
</head>
<body>
    <div class="container">
        <h1>🐍 ¡Hola desde Python CGI!</h1>
        
        <div class="info-box">
            <span class="emoji">⏰</span>
            <strong>Hora actual:</strong> """ + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")) + """
        </div>
        
        <div class="info-box">
            <span class="emoji">🌐</span>
            <strong>Método de petición:</strong> """ + os.environ.get('REQUEST_METHOD', 'Desconocido') + """
        </div>
        
        <div class="info-box">
            <span class="emoji">🔍</span>
            <strong>Query String:</strong> """ + (os.environ.get('QUERY_STRING', 'Ninguna') or 'Ninguna') + """
        </div>
        
        <div class="info-box">
            <span class="emoji">🏠</span>
            <strong>Tu navegador:</strong> """ + os.environ.get('HTTP_USER_AGENT', 'Desconocido')[:50] + """...
        </div>
        
        <div class="info-box" style="text-align: center; margin-top: 30px;">
            <a href="/cgi.html">🔙 Volver a las pruebas</a>
        </div>
    </div>
</body>
</html>""")
